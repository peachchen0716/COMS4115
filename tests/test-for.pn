for (int a = 0; a < 5; a++) {
	print(a);
}

print(10000000);

int b = 5;
for (b ; b >= 0; b--) {
	print(b);
}