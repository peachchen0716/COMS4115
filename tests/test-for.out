; ModuleID = 'Pyni'
source_filename = "Pyni"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.3 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.4 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.5 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.6 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.7 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.8 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.9 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.10 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.11 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.12 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.13 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.14 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.15 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.16 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.17 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.18 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1

declare i32 @printf(i8*, ...)

define i32 @main() {
entry:
  %a = alloca i32
  store i32 0, i32* %a
  br label %while

while:                                            ; preds = %while_body, %entry
  %a1 = load i32, i32* %a
  %normal_binop = icmp slt i32 %a1, 5
  br i1 %normal_binop, label %while_body, label %while_end

while_body:                                       ; preds = %while
  %a2 = load i32, i32* %a
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.4, i32 0, i32 0), i32 %a2)
  %a3 = load i32, i32* %a
  %uniop = add i32 %a3, 1
  store i32 %uniop, i32* %a
  br label %while

while_end:                                        ; preds = %while
  %printf4 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.8, i32 0, i32 0), i32 10000000)
  %b = alloca i32
  store i32 5, i32* %b
  %b5 = load i32, i32* %b
  br label %while6

while6:                                           ; preds = %while_body9, %while_end
  %b7 = load i32, i32* %b
  %normal_binop8 = icmp sge i32 %b7, 0
  br i1 %normal_binop8, label %while_body9, label %while_end14

while_body9:                                      ; preds = %while6
  %b10 = load i32, i32* %b
  %printf11 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.15, i32 0, i32 0), i32 %b10)
  %b12 = load i32, i32* %b
  %uniop13 = sub i32 %b12, 1
  store i32 %uniop13, i32* %b
  br label %while6

while_end14:                                      ; preds = %while6
  ret i32 0
}
